### ISSUE TITLE
[Feature Request] Soft Body Physics Support

### ISSUE THREAD

**Issue by ssandif** [xenko3d/xenko#445]
**Is your feature request related to a problem? Please describe.**
Bullet physics supports soft body physics, however, Xenko currently has not exposed any of this functionality in code or in the editor. From the [documentation](https://doc.xenko.com/latest/en/manual/physics/simulation.html): "The `SoftBodySupport`, `MultiThreaded`, and `UseHardwareWhenPossible` flags are currently disabled.". 

**Describe the solution you'd like**
- Soft body simulation functionality exposed in code
- Editor modifications to make using soft body physics easier

**Describe alternatives you've considered**
There are probably ways to imitate this for certain game mechanics using animations, but for a lot of cases I'd imagine that true physics support is the only way to get the desired behavior.

**Additional context**
- This enhancement probably depends upon #289 
- I'm not very familiar with Godot, but it looks like they've successfully implemented soft body support (also using Bullet), so it may be worthwhile to look at their editor interface for it. See https://docs.godotengine.org/en/3.1/tutorials/physics/soft_body.html for details.
- Depending on the state of the multi-threading and hardware acceleration features within Bullet, it may be worth adding issues and implementing the functionality of the other two placeholder physics flags (`MultiThreaded` and `UseHardwareWhenPossible`).
- This would be a nice feature to have that would help Xenko stand out (as far as I can tell neither Unity nor Unreal support soft body physics out of the box). However, I'm guessing that not many projects actually require it, so this issue should probably be considered low priority.
